scipy/sparse/coo.py

```diff
@@ -496,7 +496,11 @@ def _mul_vector(self, other):
         return result
 
     def _mul_multivector(self, other):
-        return np.hstack([self._mul_vector(col).reshape(-1,1) for col in other.T])
+        result = np.zeros((other.shape[1], self.shape[0]),
+                          dtype=upcast_char(self.dtype.char, other.dtype.char))
+        for i, col in enumerate(other.T):
+            coo_matvec(self.nnz, self.row, self.col, self.data, col, result[i])
+        return np.matrix(result.T)
```

Comment:
It returned `np.matrix` when `other` was a `np.matrix`, and `np.array` otherwise. Should I preserve that behavior, or just have it return `result.T`?